app/pages/reset-password.jsx

```diff
@@ -0,0 +1,222 @@
+          size="20"
+        />
+        <Translate
+          component="p"
+          content="resetPassword.newPasswordConfirmationLabel"
+        />
+        <input
+          ref={(input) => { this.confirmation = input; }}
+          type="password"
+          className="standard-input"
+          size="20"
+        />
+        <p>
+          <button
+            type="submit"
+            className="standard-button"
+            disabled={this.state.resetError || this.state.resetSuccess}
+          >
+            Submit
+          </button>{' '}
+
+          {this.state.inProgress &&
+            <i className="fa fa-spinner fa-spin form-help" />}
+
+          {this.state.resetSuccess &&
+            <i className="fa fa-check-circle form-help success" />}
+
+          {this.state.resetError &&
+            <Translate
+              className="form-help error"
+              component="small"
+              content="resetPassword.resetError"
+            />}
+        </p>
+      </form>
+    );
+  }
+
+  renderSubmitEmailForm() {
+    if (this.props.user) {
+      return (
+        <Translate
+          content="resetPassword.loggedInDialog"
+        />
+      );
+    }
+
+    return (
+      <form onSubmit={this.handleEmailSubmit}>
+        <Translate
+          component="p"
+          content="resetPassword.enterEmailDialog"
+        />
+        <input
+          type="email"
+          ref={(input) => { this.email = input; }}
+          required={true}
+          onChange={this.handleEmailChange}
+          className="standard-input"
+          size="50"
+        />
+        <p>
+          <button
+            type="submit"
+            className="standard-button"
+            disabled={!this.state.emailIsValid}
+          >
+            Submit
+          </button>{' '}
+
+          {this.state.inProgress &&
+            <i className="fa fa-spinner fa-spin form-help" />}
+
+          {this.state.emailSuccess &&
+            <i className="fa fa-check-circle form-help success" />}
+        </p>
+
+        {this.state.emailSuccess &&
+          <Translate
+            component="p"
+            content="resetPassword.emailSuccess"
+          />}
+
+        {this.state.emailError &&
+          <Translate
+            className="form-help error"
+            component="small"
+            content="resetPassword.emailError"
+          />}
+      </form>
+    );
+  }
+
+  render() {
+    return (
+      <div className="centered-grid">
+        {this.props.location.query && !this.props.location.query.reset_password_token &&
+          this.renderSubmitEmailForm()}
+
+        {this.props.location.query && this.props.location.query.reset_password_token && !this.state.resetSuccess &&
+          this.renderNewPasswordForm()}
+      </div>
+    );
+  }
+}
+
+ResetPasswordPage.propTypes = {
+  location: PropTypes.shape({
+    query: PropTypes.shape({
+      reset_password_token: PropTypes.string
+    })
+  }),
+  user: PropTypes.shape({})
```

Comment:
This makes it look like some PropType definitions are missing. It's ok for this to be just `PropTypes.object` if all we're doing is checking for the presence of the user. If eslint warning is bothersome, then you can put in an inline exception.